File: PowerFitControl/students/src/main/kotlin/br/com/pwrftctrl/students/data/models/Registration.kt
package br.com.pwrftctrl.students.data.models

import org.jetbrains.exposed.sql.Table
import br.com.pwrftctrl.core.data.utils.dateField
import br.com.pwrftctrl.core.data.utils.CurrentDate
import br.com.pwrftctrl.core.data.models.Users

object Registration : Table() {
  val id = integer("id").autoIncrement().entityId()
  val active = bool("active").default(true)
  val student = integer("student").references(Users.id)
  val registrationDate = dateField("reistraion_date").defaultExpression(CurrentDate())
  val startDate = dateField("start_date")
  val dueDate = dateField("end_date")
  val situation = varchar("situation", 120)
  val modality = integer("modality").references(Modality.id)
  val observation = varchar("observation", 255)
  val inDebit = bool("in_debit").default(false)
}

File: vuetify/academy-system/src/composables/data/models/Registration.ts
import IModel from "../interfaces/IModel";
type Status = "debited" | "debitor";
export default class Registration implements IModel {
	id!: string;
	active: boolean;
	student: string;
	createAt: Date;
	startDate: Date;
	endDate?: Date;
	isMonthlyPlan: boolean;
	status: Status;
	modality?: string;
	observation: string;
	constructor(data: {
		active: boolean;
		student: string;
		createAt: Date;
		startDate: Date;
		endDate?: Date;
		status: Status;
		modality?: string;
		observation: string;
		isMonthlyPlan: boolean;
	}) {
		this.active = data.active;
		this.student = data.student;
		this.createAt = data.createAt;
		this.startDate = data.startDate;
		this.endDate = data.endDate;
		this.status = data.status;
		this.modality = data.modality;
		this.observation = data.observation;
		this.isMonthlyPlan = data.isMonthlyPlan;
	}
}
